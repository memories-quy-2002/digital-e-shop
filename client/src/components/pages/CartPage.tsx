import React, { useCallback, useEffect, useState } from "react";
import { Button, Container, Modal } from "react-bootstrap";
import { Helmet } from "react-helmet";
import { BsArrowLeft, BsArrowRight } from "react-icons/bs";
import { useNavigate } from "react-router-dom";
import axios from "../../api/axios";
import { useToast } from "../../context/ToastContext";
import "../../styles/CartPage.scss";
import AsideCart from "../common/AsideCart";
import CartItem from "../common/CartItem";
import NavigationBar from "../common/NavigationBar";
import Layout from "../layout/Layout";
import CheckoutPaymentPage from "./CheckoutPaymentPage";
import { useAuth } from "../../context/AuthContext";

interface CartProps {
    cartItemId: number;
    productId: number;
    productName: string;
    category: string;
    brand: string;
    price: number;
    image: string;
    quantity: number;
}

const CartPage = () => {
    const navigate = useNavigate();
    const { uid } = useAuth();
    const { addToast } = useToast();
    const [cart, setCart] = useState<CartProps[]>([]);
    const [totalPrice, setTotalPrice] = useState<number>(0);
    const [discount, setDiscount] = useState<number>(0);
    const [subtotal, setSubtotal] = useState<number>(0);
    console.log(totalPrice, discount);
    const [error, setError] = useState<string>("");
    const [show, setShow] = useState<boolean>(false);
    const [isPayment, setIsPayment] = useState<boolean>(false);

    const handleShow = useCallback(() => setShow(true), []);
    const handleClose = useCallback(() => setShow(false), []);
    const togglePayment = useCallback(() => setIsPayment((prev) => !prev), []);
    const updateQuantity = (itemId: number, newQuantity: number) => {
        setCart((prevCart: CartProps[]) =>
            prevCart.map((item) => (item.cartItemId === itemId ? { ...item, quantity: newQuantity } : item))
        );
    };

    const handleQuantityChange = (itemId: number, event: React.ChangeEvent<HTMLInputElement>) => {
        const newQuantity = parseInt(event.target.value, 10);
        updateQuantity(itemId, newQuantity);
    };

    const handleClickPayment = useCallback(() => {
        togglePayment();
        setShow(false);
    }, [togglePayment]);

    const handleRemove = useCallback(
        async (cartItemId: number) => {
            try {
                const response = await axios.post(`/api/cart/delete`, {
                    cartItemId,
                });
                if (response.status === 200) {
                    console.log(response.data.msg);
                    setCart((cart) => cart.filter((cart) => cart.cartItemId !== cartItemId));
                    addToast("Remove Cart item", "Item removed from cart successfully");
                }
            } catch (err) {
                console.error(err);
            }
        },
        [addToast]
    );

    const applyDiscount = async (discountCode: string, price: number) => {
        try {
            const response = await axios.post(`/api/discount`, {
                discountCode,
                price,
            });
            if (response.status === 200) {
                const newPrice = response.data.newPrice;
                setDiscount(totalPrice - newPrice);
                setSubtotal(newPrice);
                setError("");
                addToast("Applying Coupon", "Coupon has been applied successfully");
                console.log(response.data.msg);
            } else if (response.status === 204) {
                setError("Discount not found");
                console.log(response.data.msg);
            }
        } catch (err) {
            console.error(err);
        }
    };

    useEffect(() => {
        const fetchCartItems = async () => {
            try {
                const response = await axios.get(`/api/cart/${uid}`);
                if (response.status === 200) {
                    const cartItems: CartProps[] = response.data.cartItems.map((item: any) => ({
                        ...item,
                        cartItemId: item.cart_item_id,
                        productId: item.product_id,
                        productName: item.product_name,
                        image: item.main_image,
                    }));
                    setCart(cartItems);
                }
            } catch (err) {
                console.error(err);
            }
        };

        if (uid) {
            fetchCartItems();
        }
    }, [uid]);

    useEffect(() => {
        const newTotalPrice = cart.reduce((acc, item) => acc + item.price * item.quantity, 0);
        setTotalPrice(newTotalPrice);
        setSubtotal(newTotalPrice - discount);
    }, [cart, discount]);

    return (
        <Layout>
            <NavigationBar />
            <Helmet>
                <title>Cart</title>
            </Helmet>
            <Container fluid className="cart__container">
                {isPayment ? (
                    <CheckoutPaymentPage
                        setIsPayment={setIsPayment}
                        cart={cart}
                        totalPrice={totalPrice}
                        discount={discount}
                        subtotal={subtotal}
                    />
                ) : (
                    <>
                        {" "}
                        <h3 className="cart__container__title">Shopping Cart</h3>
                        <main className="cart__container__box">
                            <section className="cart__container__box__main">
                                <div className="cart__container__box__main__list">
                                    {cart.map((item) => (
                                        <CartItem
                                            key={item.cartItemId}
                                            item={item}
                                            handleQuantityChange={handleQuantityChange}
                                            handleRemove={handleRemove}
                                        />
                                    ))}
                                </div>
                                <div className="cart__container__box__main__buttons">
                                    <button
                                        className="cart__container__box__main__buttons__continue"
                                        onClick={() => navigate("/")}
                                    >
                                        <BsArrowLeft /> Continue shopping
                                    </button>
                                    <button
                                        className="cart__container__box__main__buttons__purchase"
                                        onClick={handleShow}
                                    >
                                        Make purchase <BsArrowRight />
                                    </button>
                                </div>
                                <div className="cart__container__box__main__notification">
                                    Free delivery for 1-2 days
                                </div>
                            </section>
                            <AsideCart
                                totalPrice={totalPrice}
                                discount={discount}
                                subtotal={subtotal}
                                error={error}
                                applyDiscount={applyDiscount}
                            />
                        </main>
                    </>
                )}

                <Modal show={show} onHide={handleClose} animation={false}>
                    <Modal.Header closeButton>
                        <Modal.Title>Purchase Confirmation</Modal.Title>
                    </Modal.Header>
                    <Modal.Body>Are you sure to make purchase?</Modal.Body>
                    <Modal.Footer>
                        <Button variant="secondary" onClick={handleClose}>
                            Close
                        </Button>
                        <Button variant="primary" onClick={handleClickPayment}>
                            Confirm
                        </Button>
                    </Modal.Footer>
                </Modal>
            </Container>
        </Layout>
    );
};

export default CartPage;
