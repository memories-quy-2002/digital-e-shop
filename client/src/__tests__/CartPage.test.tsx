import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import CartPage from "../components/pages/CartPage";
import ToastProvider from "../context/ToastContext";

describe("CartPage", () => {
    it("matches the CartPage snapshot", async () => {
        const { asFragment } = render(
            <ToastProvider>
                <MemoryRouter>
                    <CartPage />
                </MemoryRouter>
            </ToastProvider>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
