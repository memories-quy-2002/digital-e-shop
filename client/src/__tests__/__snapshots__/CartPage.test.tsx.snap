// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartPage matches the CartPage snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="header__container"
    >
      <div
        class="header__container__info"
      >
        <div
          class="header__container__info__personal"
        >
          <div
            class="header__container__info__personal__item"
          >
            <svg
              color="white"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              style="color: white;"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M391 480c-19.52 0-46.94-7.06-88-30-49.93-28-88.55-53.85-138.21-103.38C116.91 298.77 93.61 267.79 61 208.45c-36.84-67-30.56-102.12-23.54-117.13C45.82 73.38 58.16 62.65 74.11 52a176.3 176.3 0 0 1 28.64-15.2c1-.43 1.93-.84 2.76-1.21 4.95-2.23 12.45-5.6 21.95-2 6.34 2.38 12 7.25 20.86 16 18.17 17.92 43 57.83 52.16 77.43 6.15 13.21 10.22 21.93 10.23 31.71 0 11.45-5.76 20.28-12.75 29.81-1.31 1.79-2.61 3.5-3.87 5.16-7.61 10-9.28 12.89-8.18 18.05 2.23 10.37 18.86 41.24 46.19 68.51s57.31 42.85 67.72 45.07c5.38 1.15 8.33-.59 18.65-8.47 1.48-1.13 3-2.3 4.59-3.47 10.66-7.93 19.08-13.54 30.26-13.54h.06c9.73 0 18.06 4.22 31.86 11.18 18 9.08 59.11 33.59 77.14 51.78 8.77 8.84 13.66 14.48 16.05 20.81 3.6 9.53.21 17-2 22-.37.83-.78 1.74-1.21 2.75a176.49 176.49 0 0 1-15.29 28.58c-10.63 15.9-21.4 28.21-39.38 36.58A67.42 67.42 0 0 1 391 480z"
              />
            </svg>
            (+84) 123 456 7890
          </div>
          <div
            class="header__container__info__personal__item"
          >
            <svg
              color="white"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              style="color: white;"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M464 80H48a16 16 0 0 0-16 16v320a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V96a16 16 0 0 0-16-16zM265.82 284.63a16 16 0 0 1-19.64 0L89.55 162.81l19.64-25.26L256 251.73l146.81-114.18 19.64 25.26z"
              />
            </svg>
            digital-e@gmail.com
          </div>
          <div
            class="header__container__info__personal__item"
          >
            <svg
              color="white"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              style="color: white;"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M261.56 101.28a8 8 0 0 0-11.06 0L66.4 277.15a8 8 0 0 0-2.47 5.79L63.9 448a32 32 0 0 0 32 32H192a16 16 0 0 0 16-16V328a8 8 0 0 1 8-8h80a8 8 0 0 1 8 8v136a16 16 0 0 0 16 16h96.06a32 32 0 0 0 32-32V282.94a8 8 0 0 0-2.47-5.79z"
              />
              <path
                d="m490.91 244.15-74.8-71.56V64a16 16 0 0 0-16-16h-48a16 16 0 0 0-16 16v32l-57.92-55.38C272.77 35.14 264.71 32 256 32c-8.68 0-16.72 3.14-22.14 8.63l-212.7 203.5c-6.22 6-7 15.87-1.34 22.37A16 16 0 0 0 43 267.56L250.5 69.28a8 8 0 0 1 11.06 0l207.52 198.28a16 16 0 0 0 22.59-.44c6.14-6.36 5.63-16.86-.76-22.97z"
              />
            </svg>
            123 ABC Street, HCM City
          </div>
        </div>
        <div
          class="header__container__info__auth"
        >
          <strong>
            Welcome Anonymous
          </strong>
          <div
            class="header__container__info__auth__button"
          >
            <div>
              <a
                href="/login"
              >
                Login
              </a>
              <span>
                 | 
              </span>
              <a
                href="/signup"
              >
                Signup
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="header__container__main"
      >
        <div
          class="header__container__main__brand"
        >
          <a
            class="header__container__main__brand__link navbar-brand"
            href="/"
          >
            DIGITAL-E
          </a>
        </div>
        <div
          class="header__container__main__search"
        >
          <input
            class="header__container__main__search__bar"
            id=""
            name=""
            placeholder="Search"
            type="text"
            value=""
          />
          <button>
            Search
          </button>
        </div>
        <div
          class="header__container__main__group"
        >
          <div
            class="header__container__main__group__item"
          >
            <svg
              fill="currentColor"
              height="28"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="28"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 448a32 32 0 0 1-18-5.57c-78.59-53.35-112.62-89.93-131.39-112.8-40-48.75-59.15-98.8-58.61-153C48.63 114.52 98.46 64 159.08 64c44.08 0 74.61 24.83 92.39 45.51a6 6 0 0 0 9.06 0C278.31 88.81 308.84 64 352.92 64c60.62 0 110.45 50.52 111.08 112.64.54 54.21-18.63 104.26-58.61 153-18.77 22.87-52.8 59.45-131.39 112.8a32 32 0 0 1-18 5.56z"
              />
            </svg>
            Wishlist
          </div>
          <div
            class="header__container__main__group__item"
          >
            <svg
              fill="currentColor"
              height="28"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="28"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="176"
                cy="416"
                r="32"
              />
              <circle
                cx="400"
                cy="416"
                r="32"
              />
              <path
                d="M456.8 120.78a23.92 23.92 0 0 0-18.56-8.78H133.89l-6.13-34.78A16 16 0 0 0 112 64H48a16 16 0 0 0 0 32h50.58l45.66 258.78A16 16 0 0 0 160 368h256a16 16 0 0 0 0-32H173.42l-5.64-32h241.66A24.07 24.07 0 0 0 433 284.71l28.8-144a24 24 0 0 0-5-19.93z"
              />
            </svg>
            Shopping Cart
          </div>
        </div>
      </div>
    </div>
    <nav
      class="navbar navbar-expand navbar-light bg-gray"
    >
      <div
        class="navbar navbar-nav"
      >
        <a
          class="navbar__item nav-link"
          data-rr-ui-event-key="/"
          href="/"
        >
          Home
        </a>
        <a
          class="navbar__item nav-link"
          data-rr-ui-event-key="/shops"
          href="/shops"
        >
          shops
        </a>
        <a
          class="navbar__item nav-link"
          data-rr-ui-event-key="/promotions"
          href="/promotions"
        >
          promotions
        </a>
        <a
          class="navbar__item nav-link"
          data-rr-ui-event-key="/about-us"
          href="/about-us"
        >
          about us
        </a>
        <a
          class="navbar__item nav-link"
          data-rr-ui-event-key="/contact-us"
          href="/contact-us"
        >
          contact us
        </a>
        <a
          class="navbar__item nav-link"
          data-rr-ui-event-key="/support"
          href="/support"
        >
          support
        </a>
        <a
          class="navbar__item nav-link"
          data-rr-ui-event-key="/news"
          href="/news"
        >
          news
        </a>
      </div>
    </nav>
    <div
      class="cart__container container-fluid"
    >
       
      <h3
        class="cart__container__title"
      >
        Shopping Cart
      </h3>
      <div
        class="cart__container__box"
      >
        <div
          class="cart__container__box__main"
        >
          <div
            class="cart__container__box__main__list"
          />
          <div
            class="cart__container__box__main__buttons"
          >
            <button
              class="cart__container__box__main__buttons__continue"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M15 8a.5.5 0 0 0-.5-.5H2.707l3.147-3.146a.5.5 0 1 0-.708-.708l-4 4a.5.5 0 0 0 0 .708l4 4a.5.5 0 0 0 .708-.708L2.707 8.5H14.5A.5.5 0 0 0 15 8"
                  fill-rule="evenodd"
                />
              </svg>
               Continue shopping
            </button>
            <button
              class="cart__container__box__main__buttons__purchase"
            >
              Make purchase 
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1 8a.5.5 0 0 1 .5-.5h11.793l-3.147-3.146a.5.5 0 0 1 .708-.708l4 4a.5.5 0 0 1 0 .708l-4 4a.5.5 0 0 1-.708-.708L13.293 8.5H1.5A.5.5 0 0 1 1 8"
                  fill-rule="evenodd"
                />
              </svg>
            </button>
          </div>
          <div
            class="cart__container__box__main__notification"
          >
            Free delivery for 1-2 days
          </div>
        </div>
        <div
          class="cart__container__box__aside"
        >
          <div
            class="cart__container__box__aside__box"
          >
            <div
              class="cart__container__box__aside__box__coupon"
            >
              <h4
                class="cart__container__box__aside__box__coupon__title"
              >
                Have coupon?
              </h4>
              <div
                class="cart__container__box__aside__box__coupon__input"
              >
                <input
                  id="coupon"
                  name="coupon"
                  style="text-transform: uppercase;"
                  type="text"
                />
                <button
                  type="button"
                >
                  Apply
                </button>
              </div>
              <p
                class="text-danger mb-2"
              />
            </div>
            <hr />
            <div
              class="cart__container__box__aside__box__price"
            >
              <ul
                class="cart__container__box__aside__box__price__list"
              >
                <li>
                  <p>
                    Total price: 
                  </p>
                  <p>
                    $0.00
                  </p>
                </li>
                <li>
                  <p>
                    Discount: 
                  </p>
                  <p>
                    $0.00
                  </p>
                </li>
                <li
                  style="font-weight: bold; font-size: 18px;"
                >
                  <p>
                    Subtotal: 
                  </p>
                  <p>
                    $0.00
                  </p>
                </li>
              </ul>
            </div>
            <hr />
            <div
              class="cart__container__box__aside__box__payment"
            >
              <h6>
                Payment method
              </h6>
              <div>
                <ul
                  style="list-style-type: none; display: flex; flex-direction: row; justify-content: space-around; padding: 0px;"
                >
                  <li
                    style="cursor: pointer;"
                  >
                    <svg
                      fill="currentColor"
                      height="36"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 16 16"
                      width="36"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="m8 0 6.61 3h.89a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-.5.5H15v7a.5.5 0 0 1 .485.38l.5 2a.498.498 0 0 1-.485.62H.5a.498.498 0 0 1-.485-.62l.5-2A.5.5 0 0 1 1 13V6H.5a.5.5 0 0 1-.5-.5v-2A.5.5 0 0 1 .5 3h.89zM3.777 3h8.447L8 1zM2 6v7h1V6zm2 0v7h2.5V6zm3.5 0v7h1V6zm2 0v7H12V6zM13 6v7h1V6zm2-1V4H1v1zm-.39 9H1.39l-.25 1h13.72z"
                      />
                    </svg>
                  </li>
                  <li
                    style="cursor: pointer;"
                  >
                    <svg
                      fill="currentColor"
                      height="36"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 576 512"
                      width="36"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M470.1 231.3s7.6 37.2 9.3 45H446c3.3-8.9 16-43.5 16-43.5-.2.3 3.3-9.1 5.3-14.9l2.8 13.4zM576 80v352c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V80c0-26.5 21.5-48 48-48h480c26.5 0 48 21.5 48 48zM152.5 331.2L215.7 176h-42.5l-39.3 106-4.3-21.5-14-71.4c-2.3-9.9-9.4-12.7-18.2-13.1H32.7l-.7 3.1c15.8 4 29.9 9.8 42.2 17.1l35.8 135h42.5zm94.4.2L272.1 176h-40.2l-25.1 155.4h40.1zm139.9-50.8c.2-17.7-10.6-31.2-33.7-42.3-14.1-7.1-22.7-11.9-22.7-19.2.2-6.6 7.3-13.4 23.1-13.4 13.1-.3 22.7 2.8 29.9 5.9l3.6 1.7 5.5-33.6c-7.9-3.1-20.5-6.6-36-6.6-39.7 0-67.6 21.2-67.8 51.4-.3 22.3 20 34.7 35.2 42.2 15.5 7.6 20.8 12.6 20.8 19.3-.2 10.4-12.6 15.2-24.1 15.2-16 0-24.6-2.5-37.7-8.3l-5.3-2.5-5.6 34.9c9.4 4.3 26.8 8.1 44.8 8.3 42.2.1 69.7-20.8 70-53zM528 331.4L495.6 176h-31.1c-9.6 0-16.9 2.8-21 12.9l-59.7 142.5H426s6.9-19.2 8.4-23.3H486c1.2 5.5 4.8 23.3 4.8 23.3H528z"
                      />
                    </svg>
                  </li>
                  <li
                    style="cursor: pointer;"
                  >
                    <svg
                      fill="currentColor"
                      height="36"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 512 512"
                      width="36"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zm-141.651-35.33c4.937-32.999-20.191-50.739-54.55-62.573l11.146-44.702-27.213-6.781-10.851 43.524c-7.154-1.783-14.502-3.464-21.803-5.13l10.929-43.81-27.198-6.781-11.153 44.686c-5.922-1.349-11.735-2.682-17.377-4.084l.031-.14-37.53-9.37-7.239 29.062s20.191 4.627 19.765 4.913c11.022 2.751 13.014 10.044 12.68 15.825l-12.696 50.925c.76.194 1.744.473 2.829.907-.907-.225-1.876-.473-2.876-.713l-17.796 71.338c-1.349 3.348-4.767 8.37-12.471 6.464.271.395-19.78-4.937-19.78-4.937l-13.51 31.147 35.414 8.827c6.588 1.651 13.045 3.379 19.4 5.006l-11.262 45.213 27.182 6.781 11.153-44.733a1038.209 1038.209 0 0 0 21.687 5.627l-11.115 44.523 27.213 6.781 11.262-45.128c46.404 8.781 81.299 5.239 95.986-36.727 11.836-33.79-.589-53.281-25.004-65.991 17.78-4.098 31.174-15.792 34.747-39.949zm-62.177 87.179c-8.41 33.79-65.308 15.523-83.755 10.943l14.944-59.899c18.446 4.603 77.6 13.717 68.811 48.956zm8.417-87.667c-7.673 30.736-55.031 15.12-70.393 11.292l13.548-54.327c15.363 3.828 64.836 10.973 56.845 43.035z"
                      />
                    </svg>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </div>
       
    </div>
    <div
      class="footer"
    >
      <div
        class="footer__menuGroup"
      >
        <div
          class="footer__menuGroup__menu"
        >
          <h5
            class="footer__menuGroup__menu__title"
          >
            Top Categories
          </h5>
          <div
            class="footer__menuGroup__menu__item"
          >
            <a
              href="/shops?categories=Laptops"
            >
              Laptops
            </a>
          </div>
          <div
            class="footer__menuGroup__menu__item"
          >
            <a
              href="/shops?categories=Cameras"
            >
              Cameras
            </a>
          </div>
          <div
            class="footer__menuGroup__menu__item"
          >
            <a
              href="/shops?categories=Smartphones"
            >
              Smartphones
            </a>
          </div>
          <div
            class="footer__menuGroup__menu__item"
          >
            <a
              href="/shops?categories=Graphic_Cards"
            >
              Graphic cards
            </a>
          </div>
        </div>
        <div
          class="footer__menuGroup__menu"
        >
          <h5
            class="footer__menuGroup__menu__title"
          >
            Help
          </h5>
          <div
            class="footer__menuGroup__menu__item"
          >
            Delivery
          </div>
          <div
            class="footer__menuGroup__menu__item"
          >
            Track an order
          </div>
          <div
            class="footer__menuGroup__menu__item"
          >
            Secure payments
          </div>
        </div>
        <div
          class="footer__menuGroup__menu"
        >
          <h5
            class="footer__menuGroup__menu__title"
          >
            About us
          </h5>
          <div
            class="footer__menuGroup__menu__item"
          >
            Careers
          </div>
          <div
            class="footer__menuGroup__menu__item"
          >
            Terms and Conditions
          </div>
          <div
            class="footer__menuGroup__menu__item"
          >
            Privacies and Policies
          </div>
        </div>
        <div
          class="footer__menuGroup__menu"
        >
          <h5
            class="footer__menuGroup__menu__title"
          >
            Feedback
          </h5>
          <div
            class="footer__menuGroup__menu__item"
          >
            Leave a feedback
          </div>
        </div>
        <div
          class="footer__menuGroup__menu"
        >
          <h5
            class="footer__menuGroup__menu__title"
          >
            Contact us
          </h5>
          <div
            class="footer__menuGroup__menu__item-disabled"
          >
            Email: digital-e@gmail.com
          </div>
          <div
            class="footer__menuGroup__menu__item-disabled"
          >
            Phone: (+84) 123 456 7890
          </div>
          <div
            class="footer__menuGroup__menu__item-disabled"
          >
            Address: 123 ABC Street, Ho Chi Minh City, Vietnam
          </div>
        </div>
      </div>
      <div
        class="footer__otherGroup"
      >
        <div
          class="footer__otherGroup__menu"
        >
          <h5
            class="footer__otherGroup__menu__title"
          >
            Find a store
          </h5>
          <div
            class="footer__otherGroup__menu__item-disabled"
          >
            Find Digital-E stores near you
          </div>
          <button
            class="btn"
            style="width: 250px;"
            type="button"
          >
            Store location
          </button>
        </div>
        <div
          class="footer__otherGroup__menu"
        >
          <h5>
            Follow Us
          </h5>
          <ul>
            <li>
              <a
                href="https://www.facebook.com/yourpage"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 320 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
                  />
                </svg>
                 
                <span>
                  Facebook
                </span>
              </a>
            </li>
            <li>
              <a
                href="https://www.twitter.com/yourprofile"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
                  />
                </svg>
                 
                <span>
                  Twitter
                </span>
              </a>
            </li>
            <li>
              <a
                href="https://www.instagram.com/yourprofile"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 448 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
                  />
                </svg>
                 
                <span>
                  Instagram
                </span>
              </a>
            </li>
            <li>
              <a
                href="https://www.linkedin.com/in/yourprofile"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 448 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M100.28 448H7.4V148.9h92.88zM53.79 108.1C24.09 108.1 0 83.5 0 53.8a53.79 53.79 0 0 1 107.58 0c0 29.7-24.1 54.3-53.79 54.3zM447.9 448h-92.68V302.4c0-34.7-.7-79.2-48.29-79.2-48.29 0-55.69 37.7-55.69 76.7V448h-92.78V148.9h89.08v40.8h1.3c12.4-23.5 42.69-48.3 87.88-48.3 94 0 111.28 61.9 111.28 142.3V448z"
                  />
                </svg>
                 
                <span>
                  LinkedIn
                </span>
              </a>
            </li>
          </ul>
        </div>
        <div
          class="footer__otherGroup__menu"
        >
          <h5
            class="footer__otherGroup__menu__title"
          >
            Newsletter subscription
          </h5>
          <div
            class="footer__otherGroup__menu__item-disabled"
          >
            Receive product news and updates to your inbox
          </div>
          <div
            class="d-flex"
          >
            <input
              class="ps-2"
              id="email_subs"
              name="email_subs"
              placeholder="Your email"
              type="email"
              value=""
            />
            <button
              class="btn"
              type="button"
            >
              Go
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="toast-container bottom-0 end-0 p-3"
    style="z-index: 1; position: fixed; bottom: 0px; right: 0px;"
  />
</DocumentFragment>
`;
